### PR TITLE
feat/RSS-ECOMM 2_11/14/15_integrate registration form with commerce tools

### PR DESCRIPTION
**Pull request by makarovaiuliia**
## Overview
Please provide a brief summary of what this pull request accomplishes. Include the motivation behind these changes and the impact they aim to have on the project.

**What has been done:**
- Integration with commercetools
- shipping and billing address handling 

**Issue(s):**

- RSS-ECOMM-2_11: Integrate the registration form with a chosen authentication service
- [x] Successful user registration with the chosen authentication service 👏
- [x] Proper submission of user information to the authentication service backend 📨
- [ ] A success message is displayed to the user upon successful account creation 🥳
- [x] Verification of successful customer creation using the browser's DevTools Network tab 🕵️‍♀️

- RSS-ECOMM-2_14: Allow users to set a default address during registration
- [x] Users can set a default address while registering. 🖱️
- [x] The setting of a default address happens simultaneously with user account creation. 🔄
- [x] The default address is saved in the user profile upon successful registration. 💾
- [x] Proper integration with the commercetools API for storing default address information. 🔗
- [x] The user interface clearly distinguishes the default address selection during registration. 🔍 

- RSS-ECOMM-2_15: Enable users to select different billing and shipping addresses during registration

- [x] Users can enter separate addresses for billing and shipping during the registration process. 🛒
- [x] Users can choose to use the same address for both billing and shipping. ✅
- [x] The chosen addresses are saved in the user profile upon successful registration. 📥
- [x] The user interface clearly distinguishes between input fields for billing and shipping addresses. 📋
- [x] Proper integration with the commercetools API for storing billing and shipping address information. 🔗 

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Additional Information
- There is no message about successful registration, i think that it can be shown in modal or after loader

## Screenshots
https://drive.google.com/file/d/1r3ItShyY04JC1Xuf0s7f0BsMUykkXikN/view?usp=sharing

### CODE REVIEW

File: kickFlip-project/src/components/form/form.css [hunk 1]
```diff
@@ -11,7 +11,7 @@
 }
 
 .reg-form {
-    width: 600px;
+    width: 700px;
     display: grid;
     grid-template-columns: repeat(2, auto);
```
Comment (SvetaIlina) [inline kickFlip-project/src/components/form/form.css:16]: что бы не прыглала форма, можно еще поменять на repeat(2, 50%);

File: kickFlip-project/src/services/userSlice.ts [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+    setCookie('accessToken', response.access_token);
+    return response;
+});
+
+export const getAnonymousToken = createAsyncThunk('user/anonymousToken', async () => {
+    const response = await getAnonymousTokenApi();
+    sessionStorage.setItem('refreshToken', response.refresh_token);
     setCookie('accessToken', response.access_token);
     return response;
 });
```
Comment (SvetaIlina) [inline kickFlip-project/src/services/userSlice.ts:22]: код дублируется, может вынести в отдельную функцию?

File: kickFlip-project/src/components/form/form.css (hunk 1 above)
Comment (makarovaiuliia) [reply kickFlip-project/src/components/form/form.css:16]: Сделала так
```ts
    grid-template-columns: repeat(2, calc(50%-16px)); /* 16px - column gap */
```

File: kickFlip-project/src/services/userSlice.ts (hunk 2 above)
Comment (makarovaiuliia) [reply kickFlip-project/src/services/userSlice.ts:22]: Добавила)

```ts
export const saveTokens = (accessToken: string, refreshToken: string): void => {
    setCookie('accessToken', accessToken);
    sessionStorage.setItem('refreshToken', refreshToken);
};

```

Comment (SvetaIlina) [reply kickFlip-project/src/services/userSlice.ts:22]: скорость на высоте!

**Review by SvetaIlina — APPROVED** [review]

**Pull request merged**